neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -806,26 +806,43 @@ object Neo4jStoreSingleton : GraphStore {
         }
 
         return IO {
-            Either.monad<StoreError>()
-                    .binding {
-                        simProfiles.bind().map { simProfile ->
-                            val regionId = (regionCode ?: map[simProfile.id])
-                                    ?: ValidationError(type = simProfileEntity.name, id = simProfile.iccId, message = "SimProfile not linked to any region")
-                                            .left()
-                                            .bind()
-
-                            val simEntry = simManager.getSimProfile(
-                                    hlr = hssNameLookup.getHssName(regionId),
-                                    iccId = simProfile.iccId)
-                                    .mapLeft { NotFoundError(type = simProfileEntity.name, id = simProfile.iccId) }
-                                    .bind()
-                            org.ostelco.prime.model.SimProfile(
-                                    iccId = simProfile.iccId,
-                                    alias = simProfile.alias,
-                                    eSimActivationCode = simEntry.eSimActivationCode,
-                                    status = simEntry.status)
-                        }
-                    }.fix()
+            Either.monad<StoreError>().binding {
```

Comment:
It is. very difficult. to grasp what goes on.   It would be very good if it could ber broken up into smaller pieces/functions  with clear intent stated.  As it is both he intent and the control flow is very obscure.

Reply:
`.binding {` is moved from separate line (810 in old code) to one line (809).

General refactoring of the code is not in the scope of this PR.

Reply:
The "boyscout principle" https://deviq.com/boy-scout-rule "Leave your code better than you found it" is always active, to some extent.    If the code being reviewed  is hard to understand, a request to address that is not  "general refactoring",  it's a suggestion for how to  make this particular PR better.   The fact that the same pattern appears in multiple locations may indicate that some general refactoring may be beneficial, but as you say, this PR is not about general refactoring.